### PR TITLE
🥅  add `AppNotFound` error to all `ID` operations

### PR DESCRIPTION
**Pull request by shkurata**
The error is added to all `competency` operations.

### CODE REVIEW

**Review by SiteSeerBE — APPROVED** [review]

File: api/test/competency-mutations.spec.ts [hunk 1]
```diff
@@ -199,6 +199,38 @@ describe('createCompetency', () => {
...
+    );
+    expect(response).toHaveProperty(
+      'data.createRootCompetency.__typename',
+      'InputError',
+    );
+    expect(response.data.createRootCompetency).toMatchObject({
+      code: 'valueNotValid',
+      path: ['title'],
+    });
+  });
```
Comment (Moeriki) [inline api/test/competency-mutations.spec.ts:232]: Title says "error on framework not found" but implementation checks title.

File: api/test/competency-mutations.spec.ts [hunk 2]
```diff
@@ -445,6 +514,38 @@ describe('createRootCompetency', () => {
...
+
```
Comment (Moeriki) [inline api/test/competency-mutations.spec.ts:548]: Title says "error on framework not found" but implementation checks title.

File: api/test/competency-queries.spec.ts [hunk 3]
```diff
@@ -17,7 +17,15 @@ async function getCompetencyForIdAndLocale(
...
-          title
+          ... on QueryCompetencySuccess {
+            data {
+              title
+            }
+          }
+          ... on UserError {
+            code
+            message
+          }
```
Comment (Moeriki) [inline api/test/competency-queries.spec.ts:28]: The helper is named `getCompetencyForIdAndLocale`. I wouldn't expect it to return an error.

We really only want helpers that are lean and mean. If they handle different cases for different tests, something is wrong.

For the error cases I would stick to fetching them within the test or create a helper that specifically fetches user errors.

File: api/test/competency-queries.spec.ts [hunk 4]
```diff
@@ -36,7 +44,15 @@ async function getCompetencyFrameworkForIdAndLocale(
...
-          title
+          ... on QueryCompetencyFrameworkSuccess {
+            data {
+              title
+            }
+          }
+          ... on UserError {
+            code
+            message
+          }
```
Comment (Moeriki) [inline api/test/competency-queries.spec.ts:55]: The helper is named `getCompetencyFrameworkForIdAndLocale`. I wouldn't expect it to return an error.

We really only want helpers that are lean and mean. If they handle different cases for different tests, something is wrong.

For the error cases I would stick to fetching them within the test or create a helper that specifically fetches user errors.

**Review by Moeriki — CHANGES_REQUESTED** [review]

**Comment by Moeriki** [comment]
The App graphql is not updated which seems improbably with such heavy API changes. Make sure to generate `yarn generate` to be sure. I expect there might also be type errors in the App as a result since it depends on the Api design.

File: api/test/competency-queries.spec.ts [hunk 5]
```diff
@@ -94,7 +40,27 @@ describe('competency', () => {
...
+          competency(id: $id) {
+            ... on QueryCompetencySuccess {
+              data {
+                title
+              }
+            }
+            ... on UserError {
+              code
+              message
+            }
```
Comment (Moeriki) [inline api/test/competency-queries.spec.ts:55]: Getting both in a test is useless. We'll only receive one of the two. The success in this case.

**Review by Moeriki — CHANGES_REQUESTED** [review]

**Comment by Moeriki** [comment]
I approved but re-requested a review from @SiteSeerBE.

**Review by Moeriki — APPROVED** [review]

**Comment by Moeriki** [comment]
A few remarks for future reference

- Don't merge main into your branch. Use rebase.

- Could you use the emojis instead of the placeholders. GitHub seems to render them, but not all clients do.

**Review by Moeriki — APPROVED** [review]

File: app/src/fragments.gql [hunk 6]
```diff
@@ -1,5 +0,0 @@
-fragment BaseErrorFields on UserError {
-  code
-  message
-  path
-}
```
Comment (SiteSeerBE) [inline app/src/fragments.gql:5]: Wat was er mis met deze methode? De nieuwe methode werkt zo te zien ook, maar geeft `Unknown fragment "BaseErrorFields"` in IDE.

File: app/src/pages/manage/frameworks/[frameworkId]/[competencyId]/create-competency.vue [hunk 7]
```diff
@@ -46,12 +51,9 @@ const competency = useResult(result);
   <template v-else-if="loading">
     <div>Loading</div>
   </template>
-  <template v-else-if="competency == null">
-    <div>Not Found</div>
-  </template>
-  <template v-else>
+  <template v-else-if="competency?.__typename == 'QueryCompetencySuccess'">
```
Comment (Moeriki) [inline app/src/pages/manage/frameworks/[frameworkId]/[competencyId]/create-competency.vue:54]: The `competency` var used to be a Competency. Now it is a competency query result. I would update the variables name.

Or maybe better update the `useResult` to again return the competency. I'm not sure doing `__typename` checks in our Vue templates is such a good idea.

**Review by Moeriki — APPROVED** [review]

**Pull request merged**